app/controllers/api/UsersController.scala

```diff
@@ -29,21 +38,59 @@ class UsersController @Inject()(val messagesApi: MessagesApi,
     * @param role   The returned array will only contain users whose role matches the given role. If role is null, the
     *               users are not filtered by role.
     * @return the users matching the given filter parameters.
-    * @todo Implement controller method
+    * @todo return 400, if request contained forbidden parameters
+    * @todo return 401 if user is Not Authorized
+    * @todo filter users by role
     */
   def userList(search: Option[String], role: Option[String]): Action[AnyContent] = SecuredAction.async { implicit request =>
-    val emptyList = UsersResponse(List[UserResponseModel]())
-    Future.successful(NotImplemented(Json.toJson(emptyList)))
+    val usersFoundBySearch = if (search.isDefined) {
+      userServiceImpl.retrieve(search.get)
+    } else {
+      userServiceImpl.retrieveAll()
+    }
+
+    usersFoundBySearch.map(seq => {
+      val jsonResponse: JsValue = Json.toJson(seq.map(mapUserToUserProtocolModel))
+      // Return not implemented until role filter is implemented
+      if (role.isDefined) {
+        NotImplemented(jsonResponse)
+      } else {
+        Ok(jsonResponse)
+      }
+    })
   }
 
 
   /**
+    * Updates the requested user.
     *
-    * @param id
-    * @return
-    * @todo Add documentationt
-    * @todo Implement controller method
+    * @param id the id of the user that should be updated
+    * @return the updated user as json or an error object
     */
-  def updateUser(id: UUID): play.mvc.Result = play.mvc.Results.TODO
+  def updateUser(id: UUID): Action[JsValue] = SecuredAction.async(parse.json) { request =>
+    request.body.validate[UserProtocolModel].map { data =>
+      userServiceImpl.retrieve(id).flatMap {
+        case Some(user) => {
+          val updateUser = User(id,
+            user.loginInfo,
+            Option(data.firstName.getOrElse(user.firstName.get)),
+            Option(data.lastName.getOrElse(user.lastName.get)),
+            user.fullName,
+            Option(data.email.getOrElse(user.email.get)),
+            user.avatarURL)
+
+          userServiceImpl
+            .save(updateUser)
+            .map { u => Ok(Json.toJson(u)) }
+        }
+        case None => Future.failed(new IdentityNotFoundException("Couldn't find user"))
+      }
+    }.recoverTotal {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2300454676/issues/source?bid=3144120&fileBranchId=3191114#l88)